src/mlir/dialect/VeronaOps.cc

```diff
@@ -111,6 +133,36 @@ static LogicalResult verify(verona::ClassOp classOp)
   return success();
 }
 
+static LogicalResult verify(verona::WhileOp whileOp)
+{
+  auto& body = whileOp.body();
+
+  // While blocks must have a condition exit with a LoopExit operation.
+  bool hasExit = false;
+  for (Block& b : body.getBlocks())
+  {
+    // Until we find a loop_exit, keep searching
+    if (!hasExit)
+    {
+      for (Operation& op : b)
+      {
+        if (verona::LoopExitOp exit_op = dyn_cast<verona::LoopExitOp>(op))
+        {
+          hasExit = true;
+          break;
+        }
+      }
+    }
+    // All blocks must terminate
+    if (!b.back().isKnownTerminator())
+      return failure();
```

Comment:
I don't think this is necessary, the builtin MLIR verifier should already be taking care of this.

Reply:
Good point